### PR TITLE
Tools: create fake obstacle

### PR DESCRIPTION
**Pull request by mickeyouyou**
update this perception tool to support cyber and new perception data struct.

### CODE REVIEW

File: modules/tools/perception/extend_prediction.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            i += 1
+            if trajectory_length < min_length:
+                second_last = points[point_num - 2]
+                last_point = points[point_num - 1]
+                x_diff = last_point.path_point.x - second_last.path_point.x
+                y_diff = last_point.path_point.y - second_last.path_point.y
+                #t_diff = last_point.path_point.t - second_last.path_point.t
+                delta_diff = math.sqrt(x_diff ** 2 + y_diff ** 2)
+                cur_len = trajectory_length
+                while cur_len < min_length && cur_len != 0.0:
```
Comment (xiaoxq) [inline modules/tools/perception/extend_prediction.py]: && -> and.
Please have the code well tested.

Comment (mickeyouyou) [reply modules/tools/perception/extend_prediction.py]: Tested on the machine, missed commit

File: modules/tools/perception/extend_prediction.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            i += 1
+            if trajectory_length < min_length:
+                second_last = points[point_num - 2]
+                last_point = points[point_num - 1]
+                x_diff = last_point.path_point.x - second_last.path_point.x
+                y_diff = last_point.path_point.y - second_last.path_point.y
+                #t_diff = last_point.path_point.t - second_last.path_point.t
+                delta_diff = math.sqrt(x_diff ** 2 + y_diff ** 2)
+                cur_len = trajectory_length
+                while cur_len < min_length add cur_len != 0.0:
```
Comment (xiaoxq) [inline modules/tools/perception/extend_prediction.py]: Still wrong. add -> and
And it's not safe to compare equal on float.

File: modules/tools/perception/extend_prediction.py [hunk 3]
```diff
@@ -60,13 +60,13 @@ def extend_prediction(prediction, min_length, min_time):
                 last_point = points[point_num - 1]
                 x_diff = last_point.path_point.x - second_last.path_point.x
                 y_diff = last_point.path_point.y - second_last.path_point.y
-                #t_diff = last_point.path_point.t - second_last.path_point.t
+                t_diff = last_point.path_point.theta - second_last.path_point.theta
                 delta_diff = math.sqrt(x_diff ** 2 + y_diff ** 2)
                 cur_len = trajectory_length
-                while cur_len < min_length add cur_len != 0.0:
+                while cur_len < min_length and abs(cur_len - 0.0) > 0.000001:
```
Comment (xiaoxq) [inline modules/tools/perception/extend_prediction.py]: abs(cur_len - 0.0) -> abs(cur_len)

**Pull request merged**